src/benchmarks/micro/libraries/System.Text.Json/Serializer/ReadJson.cs

```diff
@@ -32,6 +32,14 @@ namespace System.Text.Json.Serialization.Tests
     [AotFilter("Currently not supported due to missing metadata.")]
     public class ReadJson<T>
     {
+#if NET6_0_OR_GREATER
+        [Params(SystemTextJsonSerializationMode.Reflection, SystemTextJsonSerializationMode.SourceGen)]
+#else
+        [Params(SystemTextJsonSerializationMode.Reflection)]
+#endif
+        public SystemTextJsonSerializationMode Mode;
```

Comment:
I totally understand why you want to benchmark both modes, I just want to let you know that adding new param/argument changes the benchmark ID. Example:

```diff
- System.Text.Json.Serialization.Tests.ReadJson<Int32>.DeserializeFromReader
+ System.Text.Json.Serialization.Tests.ReadJson<Int32>.DeserializeFromReader(Mode: Reflection)
```

The Reporting System is going to treat these two benchmarks as different benchmarks:

![image](https://user-images.githubusercontent.com/6011991/204016560-e3bd0d48-ec09-4efc-8690-79234ced530f.png)

An alternative would be to add new benchmarks with "_SourceGen" or another similar suffix. Then the ID for the existing benchmark would remain the same.

Currently this type is excluded for AOT runs: 

https://github.com/dotnet/performance/blob/45e6a44504a8ce489cf10cbefd1f6d2b4932128e/src/benchmarks/micro/libraries/System.Text.Json/Serializer/ReadJson.cs#L32

If we introduce new benchmark methods rather than `params` we could apply this attribute only to the old benchmarks and hence enable the new ones for AOT:

```cs
[Benchmark]
[AotFilter("Currently not supported due to missing metadata.")]
public T DeserializeFromString()

[Benchmark]
public T DeserializeFromString_SourceGen()
```

Reply:
We're ok wit this happening now, we're not tracking any performance regressions at the moment and the test matrix is easier to maintain with the parametric approach.